taxonomy/emsi_client.py

```diff
@@ -0,0 +1,241 @@
+
+import logging
+import json
+from functools import wraps
+
+from requests.exceptions import ConnectionError, Timeout  # pylint: disable=redefined-builtin
+from slumber.exceptions import HttpNotFoundError, SlumberBaseException
+import requests
+from edx_rest_api_client.client import EdxRestApiClient
+from edx_django_utils.cache import get_cache_key, TieredCache
+
+logger = logging.getLogger(__name__)
+
+
+JOBS_QUERY_FILTER = {
+    "filter": {
+        "when": {
+            "start": "2020-01",
+            "end": "2020-06",
+            "type": "active"
+        },
+        "posting_duration": {
+            "lower_bound": 0,
+            "upper_bound": 90
+        }
+    },
+    "rank": {
+        "by": "unique_postings",
+        "min_unique_postings": 1000,
+        "limit": 20
+    },
+    "nested_rank": {
+        "by": "significance",
+        "limit": 10
+    }
+}
+
+SALARIES_QUERY_FILTER = {
+    "filter": {
+        "when": {
+            "start": "2020-01",
+            "end": "2020-06",
+            "type": "active"
+        },
+        "posting_duration": {
+            "lower_bound": 0,
+            "upper_bound": 90
+        }
+    },
+    "rank": {
+        "by": "unique_postings",
+        "min_unique_postings": 1000,
+        "limit": 20,
+        "extra_metrics": [
+            "median_posting_duration",
+            "median_salary",
+            "unique_companies"
+        ]
+    }
+}
+
+
+class JwtEMSIApiClient(object):
+    """
+    EMSI client authenticates using a access token for the given user.
+    """
+    ACCESS_TOKEN_URL = "https://auth.emsicloud.com/connect/token"
```

Comment:
We would need to add this to settings, If you think that would be done later then just add a TODO here.